source/playbooks/CIS/lambda/cis120.py

```diff
@@ -0,0 +1,206 @@
+        # send an error and exit
+        LOGGER.debug('CIS 1.20: incorrect custom action selection')
+        APPLOGGER.add_message('CIS 1.20: incorrect custom action selection')
+        return
+
+    if (cis_data['ruleid'] not in ['1.20']):
+        # Not an applicable finding - does not match rule
+        # send an error and exit
+        LOGGER.debug('CIS 1.20: incorrect custom action selection')
+        APPLOGGER.add_message('CIS 1.20: incorrect custom action selection')
+        return
+
+    #==========================================================================
+    message['AffectedObject'] = AFFECTED_OBJECT
+    try:
+        sess = BotoSession(finding.account_id, LAMBDA_ROLE)
+        iam = sess.client('iam')
+    except Exception as e:
+        LOGGER.error(e)
+        failed()
+        return
+
+    # Mark the finding NOTIFIED while we remediate
+    message['State'] = 'INITIAL'
+    notify(finding, message, LOGGER, cwlogs=APPLOGGER)
+
+    try:
+        resource = finding.details['Resources'][0]
+        id_parts = resource['Id'].split(':')
+        account = id_parts.pop()
+
+        aws_support_policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": "sts:AssumeRole",
+                    "Principal": {
+                        "AWS": f"arn:aws:iam::{account}:root"
+                    }
+                }
+            ]
+        }
+
+        role_name = 'aws_incident_support_role'
+
+        if does_role_exist(iam, role_name):
+            lowercase_str = uuid.uuid4().hex
+            role_name = f'{role_name}_{lowercase_str[0:8]}'
+
+        iam.create_role(
+            RoleName=role_name,
+            AssumeRolePolicyDocument=json.dumps(aws_support_policy),
+            Description='Created by SHARR security hub remediation 1.20 rule',
+            Tags=[
+                {
+                    'Key': 'Name',
+                    'Value': 'CIS 1.20 aws support access role'
+                },
+            ]
+        )
+
+        iam.attach_role_policy(
+            RoleName=role_name,
+            PolicyArn='arn:aws:iam::aws:policy/AWSSupportAccess',
+        )
+
+        message['State'] = 'RESOLVED'
+        message['Note'] = '' # use default
+        notify(finding, message, LOGGER, cwlogs=APPLOGGER, sns=AWS)
+        return
+
+    except Exception as e:
+        LOGGER.error(e)
+        failed()
+        return
+
+
+def does_role_exist(iam, role_name):
+    """Check if the role name exists.
+
+        Parameters
+        ----------
+        iam: iam client, required
+        role_name: string, required
+
+        Returns
+        ------
+            bool: returns if the role exists
+        """
+    role_exists = False
+
+    try:
+        response = iam.get_role(
+            RoleName=role_name
+        )
+
+        if 'Role' in response:
+            print('Yes exists')
```

Comment:
Should probably change this to `LOGGER.debug()`

Reply:
This was for debgging. Will remove, forgot to remove.